pkg/operator/ceph/cluster/cluster.go

```diff
@@ -595,6 +600,47 @@ func (c *cluster) reportTelemetry() {
 	telemetry.ReportKeyValue(c.context, c.ClusterInfo, telemetry.ExternalModeEnabledKey, strconv.FormatBool(c.Spec.External.Enable))
 }
 
+func (c *cluster) generateMonAssimilateConf(clientName string, settings map[string]string) error {
+	root := path.Join(c.context.ConfigDir, c.Namespace)
+	assimilateConfPath := path.Join(root, fmt.Sprintf("%s.txt", clientName))
+	if err := os.MkdirAll(root, 0744); err != nil {
+		return errors.Wrapf(err, "failed to create assimilateConfPath for %s.", clientName)
+	}
+
+	cfg := `
+[` + clientName + `]
+`
+	err := ioutil.WriteFile(assimilateConfPath, []byte(cfg), 0700)
+	if err != nil {
+		rook.TerminateFatal(errors.Wrapf(err, "failed to write config file"))
+	}
+	configFile := ini.Empty()
+	s, err := configFile.NewSection(clientName)
+	if err != nil {
+		return err
+	}
+
+	for key, val := range settings {
+		if _, err := s.NewKey(key, val); err != nil {
+			return errors.Wrapf(err, "failed to add key %s", key)
+		}
+	}
+	if err := configFile.SaveTo(assimilateConfPath); err != nil {
+		return errors.Wrapf(err, "failed to save config file %s", assimilateConfPath)
+	}
+
+	args := []string{"config", "assimilate-conf", "-i", assimilateConfPath, "-o", fmt.Sprintf("%s.out", assimilateConfPath)}
+	cephCmd := client.NewCephCommand(c.context, c.ClusterInfo, args)
+	out, err := cephCmd.RunWithTimeout(pkgexec.CephCommandsTimeout)
+	if err != nil {
+		logger.Errorf("failed to run command ceph %s", args)
+		return errors.Wrapf(err, "failed to set ceph config in the centralized mon configuration database; "+
+			"you may need to use the rook-config-override ConfigMap. output: %s", string(out))
+	}
```

Comment:
How about logging the contents of the output file? It there is some failure, it would list the failed settings IIUC.

Reply:
right, it will log output file when setting is not applied.